gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -210,14 +212,41 @@ private class ContentReadChannel {
     // in-place seeks.
     private byte[] skipBuffer = null;
     private ReadableByteChannel byteChannel = null;
+    // Keeps track of distance between last 2 consecutive request.
+    private LimitedFifoQueue<Long> requestDistance = new LimitedFifoQueue<Long>(2);
+    // Keeps track of last index of last served Request.
+    private long servedRequestLastIndex = -1;
     private boolean randomAccess;
+    private boolean sequentialAccess;
```

Comment:
Why we need both randomAccess and sequentialAccess?

Reply:
Yeah, they are mostly used contrasting to each other. Let me see if this can be simplified. Shouldn't be complex.